src/HonzaBotner.Discord.Services/Commands/MemberCommands.cs

```diff
@@ -266,5 +263,27 @@ public async Task CountAll(CommandContext ctx)
 
             await ctx.Channel.SendMessageAsync(count.ToString());
         }
+
+        private class NotModOrTeacherException : Exception, IRequireModAttribute
+        {
+            public Task<bool> ExecuteCheckAsync(CommandContext ctx, bool help) => throw new NotImplementedException();
+        }
+
+        private bool CanMemberUseCount(DiscordMember member)
+        {
+            IReadOnlyCollection<ulong> memberCountRoleAllowlist = new List<ulong>
+            {
+                _commonCommandOptions.ModRoleId, _commonCommandOptions.TeacherRoleId
+            };
+
+            return memberCountRoleAllowlist.Any(role => member.Roles.Select(r => r.Id).Contains(role));
+        }
```

Comment:
Recommend making it a custom command attribute instead, so that we can use it elsewhere too and not only in this class. Also attributes are supported by the lib, and the command itself shouldn't show up for people who won't pass that attribute.

Reply:
👍